src/braket/circuits/circuit.py

```diff
@@ -196,6 +199,27 @@ def qubits(self) -> QubitSet:
         """QubitSet: Get a copy of the qubits for this circuit."""
         return QubitSet(self._moments.qubits.union(self._qubit_observable_set))
 
+    @property
+    def is_parameterized(self) -> bool:
+        """
+        This is a check to see if the circuit has any free parameters.
+        This can be used to block executions of parameterized circuits without
+        iterating through the circuit each attempted execution.
+
+        Returns:
+            bool: True if the circuit has any free parameters.
+        """
+        return self._parameterized
+
+    @property
+    def parameters(self) -> set:
+        """
+        Gets a set of the parameters in the Circuit.
+        Returns:
+            set: A set of FreeParameters in the Circuit.
```

Comment:
Narrow down the signature (even if Python doesn't enforce it)
```suggestion
    def parameters(self) -> Set[FreeParameter]:
        """
        Gets a set of the parameters in the Circuit.
        
        Returns:
            Set[FreeParameter]: The `FreeParameters` in the Circuit.
```